src/Sparrow.Server/Collections/Persistent/BinaryTree.cs

```diff
@@ -54,7 +54,7 @@ public bool HasValue
         private ushort FreeNodes
         {
             get { return MemoryMarshal.Read<ushort>(_storage); }
-            set { MemoryMarshal.Write(_storage, ref value); }
+            set { MemoryMarshal.Write(_storage, in value); }
```

Comment:
That... looks pretty bad. You are intentionally violating the spirit, no?

Reply:
This is the warning in .NET 8 when 'ref' is used:
```
warning CS9191: The 'ref' modifier for argument 2 corresponding to 'in' parameter is equivalent to 'in'. Consider using 'in' instead.
```

Reply:
Nope, you are telling `MemoryMarshal.Write` to write but dont touch value. It is semantically correct, `in` is actually `readonly ref`